manifests/components/kibana.jsonnet

```diff
@@ -38,18 +39,69 @@ local strip_trailing_slash(s) = (
     },
   },
 
+  // List of Kibana plug-ins to install
+  plugins:: {
+    /* Example:
+    logtrail: {
+      version: "v0.1.31",
+      url: "https://github.com/sivasamyk/logtrail/releases/download/v0.1.31/logtrail-6.7.1-0.1.31.zip",
+    },
+    */
+  },
+
   es: error "elasticsearch is required",
 
   serviceAccount: kube.ServiceAccount($.p + "kibana") + $.metadata {
   },
 
+  pvc: kube.PersistentVolumeClaim($.p + "kibana-plugins") + $.metadata {
+    storage: "15Gi",
+  },
+
   deploy: kube.Deployment($.p + "kibana") + $.metadata {
     spec+: {
       template+: {
         spec+: {
           securityContext: {
             fsGroup: 1001,
           },
+          volumes_+: {
+            plugins: kube.PersistentVolumeClaimVolume($.pvc),
+          },
+          initContainers_+: {
+            elasticsearch_plugins_install: kube.Container("elasticsearch-plugins-install") {
+              image: KIBANA_IMAGE,
+              securityContext: {
+                allowPrivilegeEscalation: false,
+              },
+              env_+: {
+                local plugins = [{name: name} + $.plugins[name] for name in std.objectFields($.plugins)],
+                KIBANA_PLUGINS: std.join('\n', ["%(name)s,%(version)s,%(url)s" % plugin for plugin in plugins]),
+              },
+              command: [
+                "/bin/bash",
+                "-c",
+                |||
+                  set -e
+                  set -x
+                  exec &> >(tee /opt/bitnami/kibana/plugins/log)
+                  rm -rf /opt/bitnami/kibana/plugins/lost+found
```

Comment:
Since this directory will be automatically re-created I'm not sure that it is worth to be constantly deleting it.

Reply:
I dunno. I copied the code from the Helm chart. I believe it is deleted because being a directory, KIbana would get confused and believe it is a plug-in.